src/sentry/tasks/unmerge.py

```diff
@@ -520,17 +519,26 @@ def unmerge(
 
     # We fetch the events in descending order by their primary key to get the
     # best approximation of the most recently received events.
-    queryset = Event.objects.filter(project_id=project_id, group_id=source_id).order_by("-id")
 
-    if cursor is not None:
-        queryset = queryset.filter(id__lt=cursor)
+    conditions = []
+    if last_event is not None:
+        conditions.extend(
+            [
+                ["timestamp", "<=", last_event.timestamp],
+                [["timestamp", "<", last_event.timestamp], ["event_id", "<", last_event.event_id]],
```

Comment:
I'm using event_id only to handle the cases of events with the same timestamp to make sure each one gets processed once only. If we only do timestamp < last_event.timestamp otherwise we might inadvertently skip merging events with a same timestamp

Reply:
Ok, then please add a comment explaining why the query is done this way. I would also consider being explicit on the order by clause (since that is what makes this work).

Reply:
Sorry for being pedantic. I would add even more details, like an example of what could go wrong there. Imagine you are reading this code for the first time.
Something like answering these questions:
- why does this ensure events would not be processed twice
- why does this ensure all events are processed (saying "stable order" is vague. It does not capture the culprit of the problem)
- why is it not possible that a new event comes in and fit into a chunk of the list you already processed.